homeassistant/components/yeelight/config_flow.py

```diff
@@ -70,10 +73,9 @@ async def async_step_pick_device(self, user_input=None):
         if user_input is not None:
             unique_id = user_input[CONF_DEVICE]
             self._capabilities = self._discovered_devices[unique_id]
-            return self.async_create_entry(
-                title=self._async_default_name(),
-                data={CONF_ID: unique_id},
-            )
+            await self.async_set_unique_id(unique_id)
```

Comment:
Why do we even allow picking devices? It seems to me that we should just show all discovered devices without needing to ask the user to pick. There is no auth necessary. 

For a future PR, we should consider updating this config flow to support:
 - "discovery mode", it will just add all lights it can find
 - manual mode, user has to put in an IP address

Reply:
Only because you use discovery mode, it doesn't mean you want to add all devices it can find. Consider shared households with shared Wi-Fi connections, like living communities. Picking is still valid, imho.

Reply:
Picking is indeed valid. And that's why Home Assistant has features to enable and disable entities. We should not replicate that in a config flow.

Reply:
I don't want to start a principle discussion about it :) Especially not in a pending PR I'm also waiting for, plus not with somebody having "a little" bit more HA experience with me ;)
However, I would be really annoyed if it just adds everything and I need to delete/disable entities after it again. It's just against my inner will to have something added, what I don't actually want and I have to clean up after.

Again: I don't want to block this here and continue the discussion :) Just my two cents on this suggestion!

Reply:
FWIW, I gave up on the Unifi integration forever because it adds every device on my network (50+), when all I want to monitor is two devices (phones). Maybe deleting entities has gotten easier, but a config flow that results in 95% garbage that needs to be cleaned up afterwards is not a good flow. I think there should be some flexibility here.

For now, though, agree that just adding all discovered lights is better than having all broken lights.

Reply:
There was some discussion about this in the original PR adding config flow to yeelight (#37191). Finally we decided to give each device a config entry since it is more intuitive and easier to implement (and maintain).